backend/de.metas.manufacturing/src/main/java/org/eevolution/model/validator/PP_Product_Planning.java

```diff
@@ -54,4 +65,28 @@ void updateStorageAttributesKey(@NonNull final I_PP_Product_Planning productPlan
 		final AttributesKey attributesKey = AttributesKeys.createAttributesKeyFromASIStorageAttributes(asiId).orElse(AttributesKey.NONE);
 		productPlanning.setStorageAttributesKey(attributesKey.getAsString());
 	}
+
+	@ModelChange(timings = ModelValidator.TYPE_BEFORE_CHANGE,
+			ifColumnsChanged = {I_PP_Product_Planning.COLUMNNAME_M_Product_ID, I_PP_Product_Planning.COLUMNNAME_PP_Product_BOMVersions_ID})
+	public void validateBOMVersionsAndProductId(final I_PP_Product_Planning planning)
+	{
+		final ProductId productId = ProductId.ofRepoIdOrNull(planning.getM_Product_ID());
+		if (productId == null)
+		{
+			return;
+		}
+
+		Optional.ofNullable(ProductBOMVersionsId.ofRepoIdOrNull(planning.getPP_Product_BOMVersions_ID()))
+				.map(bomVersionsDAO::getBOMVersions)
+				.ifPresent(bomVersions -> {
+					if (bomVersions.getM_Product_ID() != productId.getRepoId())
+					{
+						throw new AdempiereException("PP_Product_Planning.PP_Product_BOMVersions_ID.M_Product_ID and PP_Product_Planning.M_Product_Id don't match!")
```

Comment:
:speaking_head: EN `The planning's product needs to match the BOM's product` => DE `Das Plandaten-Produkt mit mit dem Stücklistenprodukt übereinstimmen`